### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by kumarabd**
Signed-off-by: kumarabd <abishekkumar92@gmail.com>

**Description**
- re-arranged folder structures
- moved utility functions to meshkit
- added adapter versioning

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         run: |
           export DOCKER_BUILDKIT=0
           export COMPOSE_DOCKER_CLI_BUILD=0
-          docker build --no-cache -t ${{ secrets.IMAGE_NAME }}:edge-latest .
+          docker build --no-cache -t ${{ secrets.IMAGE_NAME }}:edge-latest --build-arg TOKEN=${{ secrets.GLOBAL_TOKEN }} --build-arg GIT_COMMITSHA=${GITHUB_SHA::7} --build-arg VERSION="edge-latest" .
```
Comment (leecalcote) [inline .github/workflows/build-and-release.yml:28]: This is great. Much needed in order to get version info into the adapter.

File: consul/operations.go [hunk 2]
```diff
@@ -98,15 +90,19 @@ func (h *Consul) ApplyOperation(ctx context.Context, request adapter.OperationRe
...
+				})
 				if err1 != nil {
 					h.StreamErr(&adapter.Event{
 						Operationid: request.OperationID,
 						Summary:     fmt.Sprintf("Unable to retrieve service endpoint for the service %s.", svc),
 						Details:     err1.Error(),
 					}, err1)
 				} else {
-					msg := fmt.Sprintf("%s Service endpoint %s at %s:%v", e.Summary, endpoint.Name, endpoint.Address, endpoint.Port)
+					msg := fmt.Sprintf("%s Service endpoint %s at %s:%v", e.Summary, endpoint.Name, endpoint.External.Address, endpoint.External.Port)
```
Comment (leecalcote) [inline consul/operations.go:105]: Happy to see this. This overcomes last weekend's high priority issue.

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
Lint issue:

```
Error: ineffectual assignment to `status` (ineffassign)
```

**Comment by leecalcote** [comment]
@mgfeller near as I can tell, the final namespace cleanup test is what is failing in the BATS tests on this PR. Is this accurate?

**Review by leecalcote — APPROVED** [review]

**Comment by mgfeller** [comment]
> @mgfeller near as I can tell, the final namespace cleanup test is what is failing in the BATS tests on this PR. Is this accurate?

@leecalcote yes, looks like it, the `consul` namespace itself

**Pull request merged**